### PR TITLE
GUI visual hit indicator

### PR DESCRIPTION
**Pull request by TimTheToad**
## Purpose
  - The purpose of this PR is to visualize a hit when you hit an enemy.  
When the enemy is hit the red crosshair gets displayed inside the regular crosshair.
image

## Changes
 - Added a new class called EnemyHitIndicatorGUI which invokes the storyboard in EnemyHitIndicatorGUI.xaml.
 - To fix the thread problem following member variables where introduced: 
They are kind of self-explanatory. 
image

The problem is that the ProjectileHitEvent is called from a thread that NoesisGUI dislikes
image

## Testing (Optional)
This implementation has been tested on multiplayer with a server and two clients, as well as in single Player. 

## Extra-Extra
There is currently a bug within our NoesisGUI integration. This bug makes it so that we cannot handle opacity changes in Images. I currently don't know how to fix this issue as my knowledge with the render system and the NoesisGUI integration is very limited.

### CODE REVIEW

File: CrazyCanvas/Include/GUI/HUDGUI.h [hunk 1]
```diff
@@ -63,8 +64,8 @@ class HUDGUI : public Noesis::Grid
 private:
 	GameGUIState m_GUIState;
 
-	Noesis::Border* m_pWaterAmmoRect;		
-	Noesis::Border* m_pPaintAmmoRect;		
+	Noesis::Image* m_pWaterAmmoRect = nullptr;		
+	Noesis::Image* m_pPaintAmmoRect;
```
Comment (frozenbanana) [inline CrazyCanvas/Include/GUI/HUDGUI.h]: Why does `m_pWaterAmmoRect ` need to be nullptr but `m_pPaintAmmoRect` does not? If there is no strong motivation I suggest to set either all to Noesis resources to nullptr either here or in `Init`

File: Assets/NoesisGUI/Xaml/HUD.xaml [hunk 2]
```diff
@@ -74,16 +74,24 @@
...
+                    <GradientStop Color="#72FFFFFF" Offset="1"/>
+                    <GradientStop Color="#D6494949"/>
+                    <GradientStop Color="#7200CF38" Offset="0.496"/>
+                </LinearGradientBrush>
+            </Border.Background>
+        </Border>
+
         <TextBlock x:Name="HEALTH_DISPLAY"  Grid.Column="2" Grid.Row="3" Text="100%" FontSize="24" HorizontalAlignment="Center" VerticalAlignment="Center" Margin="0" Foreground="#FFC7144D" FontWeight="Bold" Height="31.923" Width="62.224" />
         <!--<Image Source="Test.png" Grid.Column="0" Grid.ColumnSpan="2" Grid.Row="0" Grid.RowSpan="2" Margin="0" Width="383.852" Height="392.727" VerticalAlignment="Bottom" HorizontalAlignment="Left"/>
         -->
```
Comment (frozenbanana) [inline Assets/NoesisGUI/Xaml/HUD.xaml:97]: Dead code and comments should be avoid to have in master

**Review by frozenbanana — CHANGES_REQUESTED** [review]
Overall good. Small cleanup needed.

File: Assets/NoesisGUI/Xaml/HUD.xaml (hunk 2 above)
Comment (TimTheToad) [reply Assets/NoesisGUI/Xaml/HUD.xaml:97]: Done in a separate branch.

File: CrazyCanvas/Include/GUI/HUDGUI.h (hunk 1 above)
Comment (TimTheToad) [reply CrazyCanvas/Include/GUI/HUDGUI.h]: Done

**Review by frozenbanana — APPROVED** [review]
Good!

**Pull request merged**